src/ui/common/src/components/tables/PaginatedSearchTable.tsx

```diff
@@ -125,53 +153,196 @@ export const PaginatedSearchTable: React.FC<PaginatedSearchTableProps> = ({
     return value;
   };
 
-  if (searchQuery.length > 0) {
-    filteredRows = data.data.filter((rowItem) => {
-      return shouldInclude(rowItem, searchQuery, searchColumn);
+  React.useEffect(() => {
+    if (searchQuery.length > 0) {
+      const filteredRows = data.data.filter((rowItem) => {
+        return shouldInclude(rowItem, searchQuery, searchColumn);
+      });
+
+      setRows(filteredRows);
+    } else {
+      setRows(data.data);
+    }
+  }, [searchQuery]);
+
+  React.useEffect(() => {
+    if (
+      !sortConfig.sortColumn ||
+      !sortConfig.sortColumn.name ||
+      !sortConfig.sortType
+    ) {
+      setRows(data.data);
+      return;
+    }
+
+    const sortedRows = [...rows].sort((r1, r2) => {
+      const col = sortConfig.sortColumn;
+      let v1: PaginatedSearchTableRow | PaginatedSearchTableElement = r1;
+      let v2: PaginatedSearchTableRow | PaginatedSearchTableElement = r2;
+      for (const path of col.sortAccessPath) {
+        v1 = v1[path];
+        v2 = v2[path];
+      }
+
+      if (sortConfig.sortType === 'asc') {
+        if (v1 > v2) {
+          return 1;
+        } else if (v1 < v2) {
+          return -1;
+        } else {
+          return 0;
+        }
+      } else {
```

Comment:
Another way of doing these comparisons is v1.localeCompare(v2) for ascending and v2.localeCompare(v1) for descending.

Reply:
Either way works though, and may have some caveats for special character handling.

Reply:
Neat! That's definitely cleaner — I think it uses the same logic under the hood, so I'll switch it.

Reply:
Ah, nevermind. It doesn't work on numbers, only on strings, so this won't work since our element type can have numbers. :l🙁